sycl/plugins/unified_runtime/ur/adapters/native_cpu/context.cpp

```diff
@@ -0,0 +1,107 @@
+//===--------- context.cpp - Native CPU Adapter ---------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <memory>
+#include <tuple>
+
+#include "ur/ur.hpp"
+#include "ur_api.h"
+
+#include "common.hpp"
+#include "context.hpp"
+
+UR_APIEXPORT ur_result_t UR_APICALL
+urContextCreate(uint32_t DeviceCount, const ur_device_handle_t *phDevices,
+                const ur_context_properties_t *pProperties,
+                ur_context_handle_t *phContext) {
+  std::ignore = pProperties;
+  UR_ASSERT(phDevices, UR_RESULT_ERROR_INVALID_NULL_POINTER);
```

Comment:
no need to check for these conditions with ASSERTs. They are handled by the UR validation layer.

Reply:
if you're not sure what validation is being performed in the validation layer see the [ur_valddi.cpp](https://github.com/oneapi-src/unified-runtime/blob/main/source/loader/layers/validation/ur_valddi.cpp) source file which is its implementation, and here's [urContextCreate](https://github.com/oneapi-src/unified-runtime/blob/main/source/loader/layers/validation/ur_valddi.cpp#L761) specifically.

Reply:
Thanks, we'll take a look 👍